Source/Core/VideoBackends/OGL/BoundingBox.cpp

```diff
@@ -13,58 +13,182 @@
 
 static GLuint s_bbox_buffer_id;
 
+static int s_image_mask_bounds[4] = {0, 0, 0, 0};
+static u8* s_image_mask_data = nullptr;
+static GLuint s_image_mask_texture;
+
+static float s_xscale;
+static float s_yscale;
+
 namespace OGL
 {
-void BoundingBox::Init()
+GLuint BoundingBox::image_mask_fbo;
+bool BoundingBox::image_mask_updated;
+bool BoundingBox::image_mask_cleared;
+
+void BoundingBox::SetEfbScaleChanged(float xscale, float yscale)
 {
-  if (g_ActiveConfig.backend_info.bSupportsBBox)
+  s_xscale = xscale;
+  s_yscale = yscale;
+
+  if (!g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+  {
+    image_mask_updated = false;
+
+    if (s_image_mask_data)
+      free(s_image_mask_data);
+    s_image_mask_data = reinterpret_cast<u8*>(
+        malloc(static_cast<int>(640 * xscale) * static_cast<int>(528 * yscale)));
+
+    GLint old_texture_2d;
+    glGetIntegerv(GL_TEXTURE_BINDING_2D, &old_texture_2d);
+
+    if (s_image_mask_texture)
+      glDeleteTextures(1, &s_image_mask_texture);
+    glGenTextures(1, &s_image_mask_texture);
+    glBindTexture(GL_TEXTURE_2D, s_image_mask_texture);
+    glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 0);
+    glTexImage2D(GL_TEXTURE_2D, 0, GL_R8, static_cast<int>(640 * xscale),
+                 static_cast<int>(528 * yscale), 0, GL_RED, GL_UNSIGNED_BYTE, nullptr);
+
+    GLint old_fbo;
+    glGetIntegerv(GL_DRAW_FRAMEBUFFER_BINDING, &old_fbo);
+    glBindFramebuffer(GL_FRAMEBUFFER, image_mask_fbo);
+    glFramebufferTexture(GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, s_image_mask_texture, 0);
+    glBindFramebuffer(GL_FRAMEBUFFER, old_fbo);
+
+    glBindTexture(GL_TEXTURE_2D, old_texture_2d);
+  }
+}
+
+void BoundingBox::Init(float xscale, float yscale)
+{
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
   {
     int initial_values[4] = {0, 0, 0, 0};
     glGenBuffers(1, &s_bbox_buffer_id);
     glBindBuffer(GL_SHADER_STORAGE_BUFFER, s_bbox_buffer_id);
     glBufferData(GL_SHADER_STORAGE_BUFFER, 4 * sizeof(s32), initial_values, GL_DYNAMIC_DRAW);
     glBindBufferBase(GL_SHADER_STORAGE_BUFFER, 0, s_bbox_buffer_id);
   }
+  else
+  {
+    GLint old_fbo;
+    glGetIntegerv(GL_DRAW_FRAMEBUFFER_BINDING, &old_fbo);
+    glGenFramebuffers(1, &image_mask_fbo);
+    glBindFramebuffer(GL_FRAMEBUFFER, image_mask_fbo);
+    const GLenum DrawBuffers[1] = {GL_COLOR_ATTACHMENT0};
+    glDrawBuffers(1, DrawBuffers);
+    glBindFramebuffer(GL_FRAMEBUFFER, old_fbo);
+
+    SetEfbScaleChanged(xscale, yscale);
+  }
 }
 
 void BoundingBox::Shutdown()
 {
-  if (g_ActiveConfig.backend_info.bSupportsBBox)
+  if (g_ActiveConfig.backend_info.bSupportsFragmentStoresAndAtomics)
+  {
     glDeleteBuffers(1, &s_bbox_buffer_id);
+  }
+  else
+  {
+    glDeleteFramebuffers(1, &image_mask_fbo);
+    glDeleteTextures(1, &s_image_mask_texture);
+    free(s_image_mask_data);
+    s_image_mask_data = nullptr;
+  }
 }
 
 void BoundingBox::Set(int index, int value)
```

Comment:
What part of the behavior would you like to see documented? This is the same nomenclature that all of the backends use.